### PR TITLE
Gpl 769 cherrypicked samples view

### PR DESCRIPTION
**Pull request by Chris-Friend**
Changes proposed in this pull request:

* Add a new migration to include a `cherrypicked_samples` view

Note
* some of the column names don't match exactly to the crawler/lighthouse requirements, i.e. will still need some mapping of column names in crawler/lighthouse
* I have included the `created` column so the adding filtered positive fields to legacy samples migration cherrypicked samples query can also use this view

### CODE REVIEW

**Comment by Chris-Friend** [comment]
I'm thinking it may be better to not return the event_type in the view, and instead just include in the view entries that have the `cherrypick_layout_set` or `lh_beckman_cp_destination_created` event type. As otherwise the view would include none-cherrypicked samples (with different event types)

**Comment by Chris-Friend** [comment]
A downside to this approach is that we lose some of the tests we have that ensure the cross-database JOINS work correctly. Although It may be possible to setup a test environment in this repo, and write similar tests

File: db/structure.sql [hunk 1]
```diff
@@ -47,6 +47,17 @@ CREATE TABLE `bmap_flowcell` (
   CONSTRAINT `fk_bmap_flowcell_to_study` FOREIGN KEY (`id_study_tmp`) REFERENCES `study` (`id_study_tmp`) ON DELETE RESTRICT ON UPDATE RESTRICT
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci;
 /*!40101 SET character_set_client = @saved_cs_client */;
+DROP TABLE IF EXISTS `cherrypicked_samples`;
```
Comment (pjvv) [inline db/structure.sql:50]: Why does it call `DROP TABLE`?

File: db/structure.sql [hunk 2]
```diff
@@ -168,16 +179,16 @@ CREATE TABLE `lighthouse_sample` (
   `filtered_positive` tinyint(1) DEFAULT NULL COMMENT 'Filtered positive result value',
   `filtered_positive_version` varchar(255) CHARACTER SET utf8 COLLATE utf8_unicode_ci DEFAULT NULL COMMENT 'Filtered positive version',
   `filtered_positive_timestamp` datetime DEFAULT NULL COMMENT 'Filtered positive timestamp',
-  `lh_sample_uuid` varchar(36) COLLATE utf8_unicode_ci DEFAULT NULL COMMENT 'Sample uuid created in crawler',
-  `lh_source_plate_uuid` varchar(36) COLLATE utf8_unicode_ci DEFAULT NULL COMMENT 'Source plate uuid created in crawler',
+  `lh_sample_uuid` varchar(36) CHARACTER SET utf8 COLLATE utf8_unicode_ci DEFAULT NULL COMMENT 'Sample uuid created in crawler',
```
Comment (pjvv) [inline db/structure.sql:182]: Do you know why these things have changed?

File: db/structure.sql [hunk 3]
```diff
@@ -455,6 +466,19 @@ CREATE TABLE `study_users` (
   CONSTRAINT `study_users_study_fk` FOREIGN KEY (`id_study_tmp`) REFERENCES `study` (`id_study_tmp`)
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci;
 /*!40101 SET character_set_client = @saved_cs_client */;
+/*!50001 DROP VIEW IF EXISTS `cherrypicked_samples`*/;
+/*!50001 SET @saved_cs_client          = @@character_set_client */;
```
Comment (pjvv) [inline db/structure.sql:470]: Along with these?

File: db/structure.sql (hunk 2 above)
Comment (Chris-Friend) [reply db/structure.sql:182]: These changes were made automatically when I called `bundle exec rails db:migrate` to apply the view migration

File: db/structure.sql (hunk 1 above)
Comment (Chris-Friend) [reply db/structure.sql:50]: Not sure. All changes to `db/structure.sql` were made as part of the `bundle exec rails db:migrate` call. Looking at the content of this file, my guess is that it attempts to drop the table to ensure there is no name conflict with the view. Note that a table with such a name did not already exist

File: db/structure.sql (hunk 3 above)
Comment (Chris-Friend) [reply db/structure.sql:470]: Again automatic as per running `bundle exec rails db:migrate`

**Comment by Chris-Friend** [comment]
It appears the build is failing as the event_warehouse db is not setup during the continuous integration build steps. This highlights the under arching problem of the dependencies. A conditional SQL statement would be ideal, but I'm not entirely sure it is possible. Going to do some more digging

File: db/structure.sql (hunk 1 above)
Comment (pjvv) [reply db/structure.sql:50]: Cool.

File: db/structure.sql (hunk 2 above)
Comment (pjvv) [reply db/structure.sql:182]: Understood.

File: db/structure.sql (hunk 3 above)
Comment (pjvv) [reply db/structure.sql:470]: Gotcha.

**Comment by pjvv** [comment]
> It appears the build is failing as the event_warehouse db is not setup during the continuous integration build steps. This highlights the under arching problem of the dependencies. A conditional SQL statement would be ideal, but I'm not entirely sure it is possible. Going to do some more digging

Yeah, it does indeed. I wonder then if it is safer and maybe more future-proof to have a separate repo which in the CI checks out both the event_warehouse and unified_warehouse and can test against them both?

**Comment by Chris-Friend** [comment]
> > It appears the build is failing as the event_warehouse db is not setup during the continuous integration build steps. This highlights the under arching problem of the dependencies. A conditional SQL statement would be ideal, but I'm not entirely sure it is possible. Going to do some more digging
> 
> Yeah, it does indeed. I wonder then if it is safer and maybe more future-proof to have a separate repo which in the CI checks out both the event_warehouse and unified_warehouse and can test against them both?

Either this, or the alternative I'm working on: allow the database to not exist, and if so return no cherrypicked samples

**Pull request closed without merging**